### PR TITLE
Remove unnecessary use of ns.follow in ansi-c/

### PR DESCRIPTION
**Pull request by tautschnig**
Only structs, unions, enums can be hidden behind tags.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/ansi-c/anonymous_member.cpp [hunk 1]
```diff
@@ -51,14 +51,16 @@ exprt get_component_rec(
...
 
     if(comp.get_name()==component_name)
     {
       return std::move(make_member_expr(struct_union, comp, ns));
     }
-    else if(comp.get_anonymous() &&
-            (type.id()==ID_struct || type.id()==ID_union))
+    else if(
+      comp.get_anonymous() &&
+      (type.id() == ID_struct || type.id() == ID_union ||
```
Comment (kroening) [inline src/ansi-c/anonymous_member.cpp]: The struct/union case should be unnecessary.

File: src/ansi-c/c_typecheck_expr.cpp [hunk 2]
```diff
@@ -636,8 +634,9 @@ void c_typecheck_baset::typecheck_expr_builtin_offsetof(exprt &expr)
           for(const auto &c : struct_union_type.components())
           {
             if(
-              c.get_anonymous() && (follow(c.type()).id() == ID_struct ||
-                                    follow(c.type()).id() == ID_union))
+              c.get_anonymous() &&
+              (c.type().id() == ID_struct || c.type().id() == ID_union ||
+               c.type().id() == ID_struct_tag || c.type().id() == ID_union_tag))
```
Comment (kroening) [inline src/ansi-c/c_typecheck_expr.cpp]: Same here

File: src/ansi-c/c_typecheck_initializer.cpp [hunk 3]
```diff
@@ -794,8 +794,10 @@ designatort c_typecheck_baset::make_designator(
               entry.type=tmp_type;
             }
             else if(
-              c.get_anonymous() && (follow(c.type()).id() == ID_struct ||
-                                    follow(c.type()).id() == ID_union) &&
+              c.get_anonymous() &&
+              (c.type().id() == ID_struct || c.type().id() == ID_union ||
+               c.type().id() == ID_struct_tag ||
```
Comment (kroening) [inline src/ansi-c/c_typecheck_initializer.cpp]: And here!

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 2c9e3f00255d7f570572f8b5d6ae9f5af59a0b38).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97031100

File: src/ansi-c/anonymous_member.cpp (hunk 1 above)
Comment (tautschnig) [reply src/ansi-c/anonymous_member.cpp]: Done.

File: src/ansi-c/c_typecheck_expr.cpp (hunk 2 above)
Comment (tautschnig) [reply src/ansi-c/c_typecheck_expr.cpp]: Done.

File: src/ansi-c/c_typecheck_initializer.cpp (hunk 3 above)
Comment (tautschnig) [reply src/ansi-c/c_typecheck_initializer.cpp]: Done.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: c017dc777a0e0baffc1996b5c04e138a98106c9e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97048990

**Review by kroening — APPROVED** [review]

**Pull request merged**